### DISCUSSION TITLE
How can I set custom key name of version in versions.properties

### DISCUSSION THREAD

**Discussion by piyush-kukadiya** [Splitties/refreshVersions#371] [Q&A]
Currently version for `exoplayer` looks like below as generated by `buildSrcLibs` task

**versions.properties**

`version.com.google.android.exoplayer..exoplayer=2.11.5`

Can I replace above key with my own like below:

`version.exoplayer=2.11.5` 

Thanks

**Comment by jmfayard** [comment]
You have to use custom rules

See https://github.com/jmfayard/refreshVersions/issues/356#issuecomment-843276157

**Reply by piyush-kukadiya** [reply]
Can you please give an example? I am confused how existing rules are defined.

**Reply by jmfayard** [reply]
```kotlin
// settings.gradle.kts
refreshVersions {
    extraArtifactVersionKeyRules(file("refreshVersions-extra-rules.txt"))
}
```

In the file `refreshVersions-extra-rules.txt`

```
org.apache.poi:poi(-*)
    ^^^^^^.^^^
```

the key generated will be `version.apache.poi`
